core/src/main/scala/org/apache/spark/sql/delta/commands/CloneTableBase.scala

```diff
@@ -0,0 +1,356 @@
+      val absolutePaths = DeltaFileOperations.makePathsAbsolute(
+        qualifiedSourceTableBasePath,
+        datasetOfNewFilesToAdd)
+      absolutePaths
+    }
+  }
+}
+
+/**
+ * Base Trait for Clone and Restore Commands
+ */
+abstract class CloneTableBase(
+    sourceTable: CloneSource,
+    tablePropertyOverrides: Map[String, String],
+    targetPath: Path)
+  extends LeafRunnableCommand with CloneTableBaseUtils
+{
+
+  import CloneTableBase._
+
+  /**
+   * Run the clone command
+   *
+   * @param spark [[SparkSession]] to use
+   * @param opName Name of the operation used in log4j logs
+   * @param deltaOperation [[DeltaOperations.Operation]] to use when commit changes to DeltaLog
+   * @param fsOptions Extra filesystem options passed to target DeltaLog.
+   * @return
+   */
+  def runInternal(
+      spark: SparkSession,
+      opName: String,
+      hdpConf: Configuration,
+      deltaOperation: DeltaOperations.Operation,
+      fsOptions: Map[String, String]): Seq[Row] = {
+    val targetFs = targetPath.getFileSystem(hdpConf)
+    val qualifiedTarget = targetFs.makeQualified(targetPath).toString
+    val qualifiedSource = {
+      val sourcePath = sourceTable.dataPath
+      val sourceFs = sourcePath.getFileSystem(hdpConf)
+      sourceFs.makeQualified(sourcePath).toString
+    }
+
+    val destinationTable = DeltaLog.forTable(spark, targetPath, fsOptions)
+
+    val txn = destinationTable.startTransaction()
+    if (txn.readVersion < 0) {
+      destinationTable.createLogDirectory()
+    }
+
+    val (datasetOfNewFilesToAdd, newMetadata, datasetOfExistingFileToRemoveOpt) = {
+      // Make sure target table is empty before running clone
+      if (txn.snapshot.allFiles.count() > 0) {
+        throw DeltaErrors.cloneReplaceNonEmptyTable
+      }
+
+      val clonedMetadata = sourceTable.metadata.copy(
+        id = UUID.randomUUID().toString,
+        name = null, // remove the name of the table during cloning
+        description = null) // remove description of table during cloning
+      (sourceTable.allFiles, clonedMetadata, Option.empty[Dataset[AddFile]])
+    }
+
+    // TODO: we have not decided on how to implement switching column mapping modes
+    //  so we block this feature for now
+    // 1. Validate configuration overrides
+    //    this checks if columnMapping.maxId is unexpected set in the properties
+    val validatedConfigurations = DeltaConfigs.validateConfigurations(tablePropertyOverrides)
+    val metadataToUpdate = newMetadata.copy(
+      configuration = newMetadata.configuration ++ validatedConfigurations)
+    // 2. Check for column mapping mode conflict with the source metadata w/ tablePropertyOverrides
+    checkColumnMappingMode(newMetadata, metadataToUpdate)
+    // 3. Checks for column mapping mode conflicts with existing metadata if there's any
+    if (txn.readVersion >= 0) {
+      checkColumnMappingMode(txn.snapshot.metadata, metadataToUpdate)
```

Comment:
Why do we need to check for column mapping mode when we are replacing the target table? Also is there a test for it, in case we need to check.

Reply:
Yeah because we allow changing table properties during replace, this checks against that.